src/merkle/simple_smt/mod.rs

```diff
@@ -79,10 +79,9 @@ impl SimpleSmt {
         // create an empty tree
         let mut tree = Self::new(depth)?;
 
-        // check if the number of leaves can be accommodated by the tree's depth; we use a min
-        // depth of 63 because we consider passing in a vector of size 2^64 infeasible.
+        // check if the number of leaves can be accommodated by the tree's depth
         let entries = entries.into_iter();
-        let max = 1 << tree.depth.min(63);
```

Comment:
This is both unnecessary and incorrect.

1. unnecessary because if it is the case we can't construct that many items, then this code is never ran
2. wrong because if it was possible to construct this many items (e.g. memory mapped files), then the check is using the invalid depth

Reply:
There is a downside of this. Which the code would panic if we had that many items. IMO that would be the correct behavior if that would ever happen.